starter-code/src/memory.js

```diff
@@ -1,15 +1,36 @@
-// var MemoryGame = function (cards) {
-//   this.cards = cards;
-// };
+var MemoryGame = function(cards) {
+  this.cards = cards;
+  this.pickedCards = [];
+  this.pairsClicked = 0;
+  this.pairsGuessed = 0;
+};
 
-// MemoryGame.prototype.shuffleCard = function (cardsArr) {
+MemoryGame.prototype.shuffleCard = function(cardsArr) {
+  //old school!!
+  for (i = 0; i < cardsArr.length; i++) {
+    var j = Math.floor(Math.random() * cardsArr.length);
+    var temp = cardsArr[i];
+    cardsArr[i] = cardsArr[j];
+    cardsArr[j] = temp;
+  }
+  return cardsArr;
+};
 
-// };
+MemoryGame.prototype.checkIfPair = function(firstCard, secondCard) {
+  this.pairsClicked++;
```

Comment:
Instead of increasing the clicked cards number, you can push them into an array when you click them (so you can access the data later, knowing which ones you clicked). Actually, you can guess the number of cards selected checking the length of the array.